src/cheri_insts.sail

```diff
@@ -89,8 +89,8 @@ function clause execute AUICGP(imm, cd) = {
 
 union clause ast = CJAL : (bits(21), regidx)
 /*!
- * Capability register *cd* is replaced with the next instruction's **PCC** and
- * sealed as a sentry. **PCC**.**address** is incremented by *imm*.
+ * Capability register *cd* is replaced with the next instruction's **PCC** unsealed.
```

Comment:
That is true. A function could be called as both a library call and a within compilation unit call with `cjalr` used for one and `cjal` used for the other, so CJAL needs to seal as return sentry with current interrupt status.

Reply:
> The reason I made cjal not set a sealed sentry for return address is because it broke https://github.com/microsoft/cheriot-rtos/blob/main/sdk/core/loader/boot.S#L145. Looks like you use the same address for both forward jump and return jump in that code.

That's a clever trick that we use in one place, which shrinks the part of the loader that we can't erase by a couple of bytes.  If it's problematic, we can rewrite that code.  

A lot of the register zeroing operations that we're doing in the exciting path through that block are not actually necessary, since we clobber the registers anyway in the next block.  We don't care about leaking things to the platform-specific early initialisation code, because it has access to the omnipotent capabilities.  

If you just comment out line 16, the test suite still passes.  

```diff
diff --git a/sdk/core/loader/boot.S b/sdk/core/loader/boot.S
index 60da6f0..f541438 100644
--- a/sdk/core/loader/boot.S
+++ b/sdk/core/loader/boot.S
@@ -13,9 +13,6 @@
     .p2align 2
     .type start,@function
 start:
-       cjal                    .Lregs_clear
-       // The common register clearing function will not zero these registers.
-       zeroRegisters   ra, sp, gp, a0
        // At this point all registers are cleared.
 #if __has_include(<platform-early_boot.inc>)
 #      include <platform-early_boot.inc>
@@ -142,7 +139,6 @@ start:
        addi                    s1, s1, -16
        csetbounds              ca0, csp, s1
 
-.Lregs_clear:
        // a0 is used to pass arguments to the scheduler entry.
        zeroAllRegistersExcept  ra, sp, gp, a0
        cjalr                   cra
```

This diff has no impact on the security (at worst, we leak capabilities or uninitialised values into code that owns the root capabilities) and should improve code size.